### PR TITLE
docs: rework section on node storage

### PR DESCRIPTION
**Pull request by BramGruneir**
This adds a little bit on repair and rebalancing as well.

Part of #9634.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9652)

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -629,27 +629,50 @@ There are some additional system entries sprinkled amongst the
...
+
+- the number of replicas per store
+- the total size of the data used per store
+- free space available per store
+
+In the future, some other factors that might be considered include:
+
+- cpu/network load per store
+- ranges that are used together often in queries
+- number of active ranges per store
```
Comment (petermattis) [inline docs/design.md]: Add `number of range leases held per store`.

File: docs/design.md [hunk 2]
```diff
@@ -629,27 +629,50 @@ There are some additional system entries sprinkled amongst the
...
+- ranges that already have some overlap (to decrease the chance of data loss)
```
Comment (petermattis) [inline docs/design.md]: Can you elaborate on this? I'm not sure what it means so it probably needs to be rephrased.

File: docs/design.md [hunk 3]
```diff
@@ -629,27 +629,50 @@ There are some additional system entries sprinkled amongst the
...
-
--   consensus metadata
-
--   split/merge activity
-
-A really good reference on tuning Linux installations with RocksDB is
-[here](http://docs.basho.com/riak/latest/ops/advanced/backends/leveldb/).
+# Stores and Storage
+
+Nodes contain one ore more stores. And each of those stores should be on a
```
Comment (petermattis) [inline docs/design.md]: `s/ ore / or /g`

File: docs/design.md [hunk 4]
```diff
@@ -629,27 +629,50 @@ There are some additional system entries sprinkled amongst the
...
+unique disk and contains an instance of RocksDB. And these stores in turn have
```
Comment (petermattis) [inline docs/design.md]: The second sentence is slightly odd. We create the RocksDB instances. Perhaps: `Each store should be placed on a unique disk. Internally, each store contains a single instance of RocksDB with a block cache shared amongst all of the stores in a node.`

File: docs/design.md [hunk 5]
```diff
@@ -629,27 +629,50 @@ There are some additional system entries sprinkled amongst the
...
+a collection of range replicas. More than one replica for a range will never
+occur on the same store or even the same node.
```
Comment (petermattis) [inline docs/design.md]: `s/occur/be placed/g`

File: docs/design.md [hunk 6]
```diff
@@ -629,27 +629,50 @@ There are some additional system entries sprinkled amongst the
...
+
+Early on, when a cluster is first initialized, the few default starting ranges
+will only have a single replica, but as soon as other nodes are available they
+will replicate to them until they've reached their desired replication factor,
+the default being 3. Ranges can have different replication factors and when set
```
Comment (petermattis) [inline docs/design.md]: Mention zone configs?

File: docs/design.md [hunk 7]
```diff
@@ -629,27 +629,50 @@ There are some additional system entries sprinkled amongst the
...
+they will up or down replicate to the appropriate number of replicas. Since
+ranges are only created via splits, all replicas for a range split at the same
+time so the replication factors rarely need to be adjusted.
```
Comment (petermattis) [inline docs/design.md]: I'm not sure how the first part of the sentence connects to the second. Why does the splitting of a range to create new ranges affect the desire to change replication factors?

File: docs/design.md [hunk 8]
```diff
@@ -629,27 +629,50 @@ There are some additional system entries sprinkled amongst the
...
+there is no quorum and the whole range will be considered unavailable until at
+least greater than 50% of the replicas are again available.
+
+# Rebalancing
+
+As more data are added to the system, some stores may grow faster than others.
+To combat this and to spread the overall load across the full cluster, replicas
+will be moved between stores maintaining the desired replication factor. The
+heuristics used to perform this rebalancing are in flux, but they take into
+account a number of different factors:
```
Comment (petermattis) [inline docs/design.md]: I'd leave out the in flux part and just describe the currently used heuristics and the load-based ones we're considering for the future.

**Comment by BramGruneir** [comment]
Made the requested changes.

File: docs/design.md (hunk 8 above)
Comment (BramGruneir) [inline docs/design.md]: done

File: docs/design.md (hunk 7 above)
Comment (BramGruneir) [inline docs/design.md]: removed

File: docs/design.md (hunk 4 above)
Comment (BramGruneir) [inline docs/design.md]: much better, done

File: docs/design.md (hunk 3 above)
Comment (BramGruneir) [inline docs/design.md]: done

**Review by petermattis — APPROVED** [review]
LGTM

**Pull request merged**